realm/realm-library/src/androidTest/kotlin/io/realm/EmbeddedObjectsTest.kt

```diff
@@ -527,21 +528,155 @@ class EmbeddedObjectsTest {
     }
 
     @Test
-    @Ignore("Add in another PR")
     fun insertOrUpdate_deletesOldEmbeddedObject() {
-        TODO()
+        realm.executeTransaction { realm ->
+            val parent = EmbeddedSimpleParent("parent")
+            val originalChild = EmbeddedSimpleChild("originalChild")
+            parent.child = originalChild
+            realm.insert(parent)
+
+            assertEquals(1, realm.where<EmbeddedSimpleChild>().count())
+            val managedChild = realm.where<EmbeddedSimpleChild>()
+                    .equalTo("childId", "originalChild")
+                    .findFirst()
+            assertTrue(managedChild!!.isValid)
+
+            val newChild = EmbeddedSimpleChild("newChild")
+            parent.child = newChild
+            realm.insertOrUpdate(parent)
+            assertTrue(!managedChild.isValid)
+
+            assertEquals(1, realm.where<EmbeddedSimpleChild>().count())
+            val managedNewChild = realm.where<EmbeddedSimpleChild>()
+                    .equalTo("childId", "newChild")
+                    .findFirst()
+            assertEquals(managedNewChild!!.childId, "newChild")
+            realm.where<EmbeddedSimpleChild>()
+                    .equalTo("childId", "originalChild")
+                    .findAll()
+                    .size
+                    .also { oldChildCount ->
```

Comment:
Funny mix of local variables (above `managedChild`, etc.) and `also`. Maybe stick to one pattern. 

I find it a bit annoying that you have to scan for the assert out of the top level hierarchy. If you want to skip the local variable a proposal could maybe be to embed the query directly in the `assert` instead of embedding the assert in the `also`-clause?